src/main/java/org/gridsuite/voltageinit/server/computation/service/AbstractWorkerService.java

```diff
@@ -0,0 +1,241 @@
+/**
+ * Copyright (c) 2024, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package org.gridsuite.voltageinit.server.computation.service;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.powsybl.commons.PowsyblException;
+import com.powsybl.commons.reporter.Reporter;
+import com.powsybl.commons.reporter.ReporterModel;
+import com.powsybl.iidm.network.Network;
+import com.powsybl.iidm.network.VariantManagerConstants;
+import com.powsybl.network.store.client.NetworkStoreService;
+import com.powsybl.network.store.client.PreloadingStrategy;
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.http.HttpStatus;
+import org.springframework.messaging.Message;
+import org.springframework.web.server.ResponseStatusException;
+
+import java.util.Map;
+import java.util.UUID;
+import java.util.concurrent.CancellationException;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.concurrent.locks.Lock;
+import java.util.concurrent.locks.ReentrantLock;
+import java.util.function.Consumer;
+
+/**
+ * @author Mathieu Deharbe <mathieu.deharbe at rte-france.com>
+ * @param <S> powsybl Result class specific to the computation
+ * @param <R> Run context specific to a computation, including parameters
+ * @param <P> powsybl and gridsuite Parameters specifics to the computation
+ * @param <T> result service specific to the computation
+ */
+public abstract class AbstractWorkerService<S, R extends AbstractComputationRunContext<P>, P, T extends AbstractComputationResultService<?>> {
+    protected static final Logger LOGGER = LoggerFactory.getLogger(AbstractWorkerService.class);
+
+    protected final Lock lockRunAndCancel = new ReentrantLock();
+    protected final ObjectMapper objectMapper;
+    protected final NetworkStoreService networkStoreService;
+    protected final ReportService reportService;
+    protected final ExecutionService executionService;
+    protected final NotificationService notificationService;
+    protected final AbstractComputationObserver<S, P> observer;
+    protected final Map<UUID, CompletableFuture<S>> futures = new ConcurrentHashMap<>();
+    protected final Map<UUID, CancelContext> cancelComputationRequests = new ConcurrentHashMap<>();
+    protected final T resultService;
```

Comment:
Generally it's better to pass to private and use getters/`@Getter`.

Reply:
I have no strong opinion on this, do as you wish. But if you add setters make them protected restricted.

Reply:
as it is used only by children classes for me it's ok to keep protected